app/models/registration_payment.rb

```diff
@@ -11,23 +11,43 @@ class RegistrationPayment < ApplicationRecord
 
   delegate :auto_accept_preference_live?, to: :registration
   after_create :auto_accept_hook, if: :auto_accept_preference_live?
-  after_create :auto_close_hook, unless: :refunded_registration_payment_id?
+  after_save :auto_close_hook
+
+  scope :completed, -> { where(is_completed: true) }
 
   monetize :amount_lowest_denomination,
            as: "amount",
            allow_nil: true,
            with_model_currency: :currency_code
 
+  def should_auto_close?
+    refunded_registration_payment_id.nil? && saved_change_to_is_completed?(to: true)
+  end
+
+  def create_uncaptured_payment
```

Comment:
Where is this method being used, and why are you creating a method in `RegistrationPayment` to create a new `RegistrationPayment`?

Reply:
I honestly have no idea - removed